### PR TITLE
Computation of the StorageVersionHash use overridden storage versions in unit test

### LINKED ISSUE #110750: Computation of the StorageVersionHash is incorrect for overridden storage versions in unit test
Storage version defaults to the preferred version for a given group, but can be overridden when constructing API server storage.

This is done here: https://github.com/kubernetes/kubernetes/blob/035f22d7810a8970e42d60f276f71a519b174a44/pkg/kubeapiserver/default_storage_factory_builder.go#L60-L67

Reasons the preferred version might be overridden:
* A given resource/kind does not yet exist in the preferred version (like when introducing a new alpha type in an existing API group with more stable versions)
* A given resource/kind is continuing to store in etcd in an older version for compatibility with n-1 API servers

The etcd integration test verifies that we're actually storing the correct versions in etcd, but the computation of the StorageVersionHash is not honoring the overridden version.

on master, this code reports v1 CSIStorageCapacity:

https://github.com/kubernetes/kubernetes/blob/035f22d7810a8970e42d60f276f71a519b174a44/staging/src/k8s.io/apiserver/pkg/endpoints/installer.go#L398-L408

even though the etcd test verifies we're still persisting v1beta1

/kind bug
/priority important-soon
/sig api-machinery

cc @pohly

_Originally posted by @liggitt in https://github.com/kubernetes/kubernetes/issues/109394#issuecomment-1164608879_

### PR DESCRIPTION
**Pull request by 249043822**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #110750

#### Special notes for your reviewer:
/cc @liggitt 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controlplane/instance_test.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func setUp(t *testing.T) (*etcd3testing.EtcdTestServer, Config, *assert.Assertio
 		batch.Resource("cronjobs").WithVersion("v1beta1"),
 		apisstorage.Resource("volumeattachments").WithVersion("v1beta1"),
 		networking.Resource("ingresses").WithVersion("v1beta1"),
+		apisstorage.Resource("csistoragecapacities").WithVersion("v1beta1"),
 	}
```
[the remark this answers is not in the document]
Comment (249043822) [reply pkg/controlplane/instance_test.go]: Make sense, done

File: pkg/controlplane/storageversionhashdata/data.go [hunk 2]
```diff
@@ -72,9 +72,9 @@ var GVRToStorageVersionHash = map[string]string{
 	"storage.k8s.io/v1/csidrivers":                                     "hL6j/rwBV5w=",
 	"storage.k8s.io/v1/csinodes":                                       "Pe62DkZtjuo=",
 	"storage.k8s.io/v1/storageclasses":                                 "K+m6uJwbjGY=",
-	"storage.k8s.io/v1/csistoragecapacities":                           "xeVl+2Ly1kE=",
-	"storage.k8s.io/v1/volumeattachments":                              "vQAqD28V4AY=",
-	"storage.k8s.io/v1beta1/csistoragecapacities":                      "xeVl+2Ly1kE=",
+	"storage.k8s.io/v1/csistoragecapacities":                           "4as6MA/kOg0=",
+	"storage.k8s.io/v1/volumeattachments":                              "tJx/ezt6UDU=",
+	"storage.k8s.io/v1beta1/csistoragecapacities":                      "4as6MA/kOg0=",
```
Comment (liggitt) [inline pkg/controlplane/storageversionhashdata/data.go:77]: 👍 verified these match reality now

**Comment by liggitt** [comment]
the change looks good, will tag once the import is tweaked

**Pull request merged**